File: src/main/java/info/dddeurope/lab/app/eventHandlers/PostPublishedEventHandler.java
package info.dddeurope.lab.app.eventHandlers;

import info.dddeurope.lab.app.dtos.PostDto;
import info.dddeurope.lab.app.events.PostPublishedEvent;
import info.dddeurope.lab.app.views.PostsIndex;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class PostPublishedEventHandler {

    PostsIndex index;

    public void handle(PostPublishedEvent event) throws Exception {
        final PostDto postDto = event.getPostDto();
        index.indexPost(event.getPostId(), postDto.getTitle(), postDto.getBody(), postDto.getPublisherId());
    }
}

File: DotNet/App/EventHandlers/PostPublishedEventHandler.cs
using App.Events;
using App.Views;

namespace App.EventHandlers
{
    public class PostPublishedEventHandler
    {
        readonly PostsIndex _index;

        public PostPublishedEventHandler(PostsIndex index)
        {
            _index = index;
        }

        public void Handle(PostPublishedEvent evt)
        {
            var postDto = evt.PostDto;
            _index.IndexPost(evt.PostId, postDto.Title, postDto.Body, postDto.PublisherId);
        }
    }
}